First run simulator with -n to ensure it always opens a device

DanTup opened the pull request:
@devoncarew This is a fix for the issue described in https://github.com/flutter/flutter-intellij/issues/2234. If you Cmd+W in Simulator to close the window but leave the app running, then `flutter emulator --launch ios` wouldn't seem to work. I found that passing `-n` which is supposed to force a new instance solves this (Simulator seems to not allow multiple running devices, so it doesn't actually create multiple instances).

However, a side-effect was that if it was already running, it bailed and didn't come to the foreground, so we just run it again without the `-n` immediately after (which brings it to the foreground).

It seems to work fine for me without having to show a message to the user. LMK what you think.

(cc @pq)

DanTup committed:
First run simulator with -n to ensure it always opens a device

pq commented on packages/flutter_tools/lib/src/ios/ios_emulators.dart:
```diff
@@ -37,11 +37,23 @@ class IOSEmulator extends Emulator {
     }
+
+    // First run with `-n` to force a device to boot if there isn't already one
+    await launchSimulator(<String>['-n']);
```
Is there any possible harm in needlessly running this?  A few questions:

1. what happens if this errors?  Will things be even more confused if we try and run it again?
2. what does the simulator do if it's *never* been run?  That is, how can it choose a default device if you have never selected one? (I assume it just uses the last choice?  What if there was none?)

**EDIT:** looking for docs now...

DanTup replied:
1. I think at worst we'd just show two errors. We could skip the second run if the first one fails though?

2. Er, not sure - I thought it opened something by default the first time I ran it. Is there a way we can clear its data to find out?

I did see that simctl seems to have options for listing and running specific devices, I did wonder whether there's anything useful in there (like allowing the user to select a specific device and then launch that) but it might get confusing if they think they can run multiple together (which I presume, they can not).

pq replied:
Yeah, I looked at `simctl`.  A bit of a rabbit hole.  In fact that's how I missed this easier solution!  Down the road adding support for selecting devices and potentially other bits that `simctl` offers sounds good but IMO not pressing.

Skipping the second run on a failure sounds good, though maybe overly defensive.

Your call!

DanTup committed:
If first call to iOS simulator fails, don't try second call

This will avoid two error messages in the case something is wrong with the simulator.

DanTup commented:
Pushed a minor change to not output two errors in the case the simulator fails to launch.

DanTup committed:
Add unsaved file...
